packages/chain-adapters/src/evm/EvmBaseAdapter.ts

```diff
@@ -238,7 +239,7 @@ export abstract class EvmBaseAdapter<T extends EvmChainId> implements IChainAdap
       value = erc20Balance
     }
 
-    const data = contractData ?? (await getErc20Data(to, value, contractAddress))
+    const data = memo || contractData || (await getErc20Data(to, value, contractAddress))
```

Comment:
Safer, memo should be undefined or a non-empty string but you never know what consumers might pass